misc/pytorch_toolkit/chest_xray_screening_optimised/utils/generate.py

```diff
@@ -0,0 +1,201 @@
+
+
+def get_best_values(alpha, beta):
+    """
+        Alpha changes the depth while beta changes the width
+    """
+    depth, block_new = new_lengths(alpha)
+    width, growth_rate, init_chan, bn_size = new_channels(
+        block_new, alpha, beta)
+    return block_new, growth_rate, init_chan, bn_size
+
+
+def give_model(alpha, beta, num_class=14, input_shape=(320, 320), verbose=1):
+    blocks, gw, b, bn_size = get_best_values(alpha, beta)
+    print(blocks, gw, b, bn_size)
+    model = modelzoo.DenseNet(
+        growth_rate=gw,
+        block_config=blocks,
+        num_init_features=b,
+        bn_size=bn_size,
+        num_classes=num_class)
+    total_mac = densenet_flops(
+        input_shape,
+        blocks,
+        gw,
+        bn_size,
+        b,
+        verbose=verbose)
+    return model, total_mac
+
+
+def last_function(alpha, beta, get_val=False):
+    blocks, gw, b, bn_size = get_best_values(alpha, beta)
+    print(blocks, gw, b, bn_size)
+    model = modelzoo.DenseNet(
+        growth_rate=gw,
+        block_config=blocks,
+        num_init_features=b,
+        bn_size=bn_size,
+        num_classes=14)
+    if get_val:
+        return get_flops(model, get_val)
+    else:
+        get_flops(model)
+
+
+def conv2d_flops(out1, out2, co, cin, k):
+    return out1 * out2 * co * cin * k * k
+
+
+def batchnorm_flops(out1, out2, c):
+    return 2 * out1 * out2 * c
+
+
+def relu_flops(out1, out2, c):
+    return out1 * out2 * c
+
+
+def maxpool2d_flops(out1, out2, co, k):
+    return out1 * out2 * co * (k**2 - 1)
+
+
+def avgpool2d_flops(out1, out2, co, k):
+    return out1 * out2 * co * (k**2)
+
+
+def dense_layer_flops(out1, out2, input_layers, growth_rate, k):
+    total = 0
+    total += batchnorm_flops(out1, out2, input_layers)
+    total += relu_flops(out1, out2, input_layers)
+    total += conv2d_flops(out1, out2, input_layers, growth_rate * k, 1)
+    cout = growth_rate * k
+    total += batchnorm_flops(out1, out2, cout)
+    total += relu_flops(out1, out2, cout)
+    total += conv2d_flops(out1, out2, growth_rate, cout, 3)
+    return total
+
+
+def dense_block_flops(out1, out2, layers, input_layers, growth_rate, k):
+    total = 0
+    for layer in range(layers):
+        in_layers = input_layers + layer * growth_rate
+        total += dense_layer_flops(out1, out2, in_layers, growth_rate, k)
+
+    return total
+
+
+def transition_flops(out1, out2, input_channels):
+    total = batchnorm_flops(out1, out2, input_channels)
+    total += relu_flops(out1, out2, input_channels)
+    total += conv2d_flops(out1, out2, input_channels, input_channels // 2, k=1)
+    total += avgpool2d_flops(out1 // 2, out2 // 2, input_channels // 2, k=2)
+    return total
+
+
+def densenet_flops(input_shape, block_config, growth_rate,
+                   k, init_features, verbose=1):
+    out1 = (input_shape[0] - 7 + 2 * 3) // 2 + 1
+    out2 = (input_shape[1] - 1) // 2 + 1
+    x = conv2d_flops(out1, out2, init_features, 3, 7)

```

Comment:
If I understand correctly, this code counts flops for a model. You can use the [ready python module](https://github.com/sovrasov/flops-counter.pytorch/blob/master/ptflops/flops_counter.py) for it.